### DISCUSSION TITLE
A milestone or release for SimRel 2022-12 M3?

### DISCUSSION THREAD

**Discussion by mickaelistria** [eclipse-m2e/m2e-core#1051] [General] [answered]
Hi,
Would it be possible to publish a new milestone of m2e from master (or maybe a release if there is enough substance, good enough quality and nothing too important pending)? m2e has started using Wild Web Developer 1.0 and newer LemMinX-Maven which should fix a bunch of things. Moreover, the version restrictions make that Wild Web Developer 1.0 cannot be pushed to SimRel until a compatible build of m2e is available.
Or maybe we can include m2e snapshot in SimRel for the moment?

**Comment by laeubi** [comment]
At first, I think we can release any time, and I do not care how it is "named", so why not a full release.

Beside that, I think it would be good to have

- https://github.com/eclipse-m2e/m2e-core/pull/1045

included I just need to find out why these tests are not happy anymore now we do the caching more correct, but thats not mandatory I think

So what would be the latest time a contribution for SimRel 2022-12 should happen?

**Comment by HannesWell** [comment, marked as answer]
I would like to try to have https://github.com/eclipse-m2e/m2e-core/pull/852 and https://github.com/eclipse-m2e/m2e-core/pull/572 (and maybe even https://github.com/eclipse-m2e/m2e-core/pull/466) in the December SimRel.

From https://wiki.eclipse.org/Category:SimRel-2022-12 I think that we should contribute the corresponding M2E release latest until 01.12, probably better a few days earlier.
Since it is only two~three weeks until then and the main users come from SimRel I think there is no great gain to have a full release now and doing a second one then.

Therefore I would prefer to have a new 'Milestone' (i.e. a fixed snapshot), but would be totally fine with that.
This would also be much less work:
The Jenkins job `https://ci.eclipse.org/m2e/job/batch%20infra%20stuff` is already prepared for that, one just has to change the target to `2.1.0-M2` (or whatever is suitable, just different to the existing `2.1.0-M1`). And then update the m2e SimRel contribution.
I can do that you others agree.

In general we could also consider to again make a initially 'hidden' release and already use the 4.26-I builds as TP so that we can bring https://github.com/eclipse-m2e/m2e-core/pull/853 into Eclipse 2022-12. The `releases/latest` would then point to the previous release until 4.26 has been released as well. The last time we had not yet released dependencies there were complaints that we should not do that for the releases/latest because some users use that as update-site. But there were no complaints that we had a named releases repo (that was not the latest) that hat not yet released deps. I wonder if it would be ok to repeat that?

**Reply by mickaelistria** [reply]
actually, I'm happy with adding the snapshots to SimRel for the moment, I don't think a milestone is useful.

**Reply by merks** [reply]
Please don't do this.  It might be easier, but it causes real problems:

https://github.com/eclipse-m2e/m2e-core/issues/1069

Those problems are disruptive for others, yet you won't notice a problem until someone tells you.
